Tahi-project/tahi
# Feature/aperta 9692 due date scheduled events

JIRA issue: https://jira.plos.org/jira/browse/APERTA-9692

#### What this PR does:

Adds the ability to schedule events around due dates

#### Major UI changes

<img width="1137" alt="aperta-9692" src="https://user-images.githubusercontent.com/390059/28842696-8a35a990-76f6-11e7-95f0-2d220478969e.png">

<img width="1439" alt="plos-ciagent-pr-3288" src="https://user-images.githubusercontent.com/390059/29094283-b1378fca-7c84-11e7-9d91-084801e43f26.png">

---

### Review instructions
Reviewing this would require you run some rake tasks. These are the steps to make sure the app is setup properly for review

1. Ensure the `REVIEW_DUE_AT` feature flags is turned on
2. Run `rake one_off:create_reviewer_report_scheduled_templates`
3. Invite a reviewer to review a paper
4. Accept the review
5. Check that the reminders are properly set in the workflow view, on the "Review by {Invited Reviewer}" card

#### Code Review Tasks:

**Author tasks** (delete tasks that don't apply to your PR, this list should be finished before code review):

- [ ] If I changed the database schema, I enforced database constraints.

**Reviewer tasks** (these should be checked or somehow noted before passing on to PO):
